android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java

```diff
@@ -0,0 +1,164 @@
+            if (userAsset.isErc721 /*|| userAsset.isNft*/) { // solana nfts are not supported yet
+                AsyncUtils.GetNftMetaDataContext nftMetaData = new AsyncUtils.GetNftMetaDataContext(
+                        nftMetaDataHandler.singleResponseComplete);
+                nftMetaData.asset = userAsset;
+                mJsonRpcService.getErc721Metadata(userAsset.contractAddress, userAsset.tokenId,
+                        userAsset.chainId, nftMetaData);
+                nftMetaDatas.add(nftMetaData);
+            }
+        }
+        nftMetaDataHandler.setWhenAllCompletedAction(() -> {
+            List<NftDataModel> nftDataModels = new ArrayList<>();
+            for (AsyncUtils.GetNftMetaDataContext metaData : nftMetaDatas) {
+                nftDataModels.add(new NftDataModel(metaData.asset, networkInfo,
+                        new Erc731MetaData(metaData.erc721Metadata, metaData.errorCode,
+                                metaData.errorMessage)));
+            }
+            _mNftModels.postValue(nftDataModels);
+        });
+    }
+
+    void resetServices(Context context, TxService mTxService, KeyringService mKeyringService,
+                       BlockchainRegistry mBlockchainRegistry, JsonRpcService mJsonRpcService,
+                       EthTxManagerProxy mEthTxManagerProxy,
+                       SolanaTxManagerProxy mSolanaTxManagerProxy,
+                       BraveWalletService mBraveWalletService,
+                       AssetRatioService mAssetRatioService) {
+        synchronized (mLock) {
+            mContext = context;
+            this.mTxService = mTxService;
+            this.mKeyringService = mKeyringService;
+            this.mBlockchainRegistry = mBlockchainRegistry;
+            this.mJsonRpcService = mJsonRpcService;
+            this.mEthTxManagerProxy = mEthTxManagerProxy;
+            this.mSolanaTxManagerProxy = mSolanaTxManagerProxy;
+            this.mBraveWalletService = mBraveWalletService;
+            this.mAssetRatioService = mAssetRatioService;
+        }
+    }
+
+    public class NftDataModel {
+        public BlockchainToken token;
+        public NetworkInfo networkInfo;
+        public Erc731MetaData erc731MetaData;
+
+        public NftDataModel(BlockchainToken token, NetworkInfo networkInfo,
+                            Erc731MetaData erc731MetaData) {
+            this.token = token;
+            this.networkInfo = networkInfo;
+            this.erc731MetaData = erc731MetaData;
+        }
+    }
+
+    public class Erc731MetaData implements Serializable {
+        public String mDescription;
+        public String mImageUrl;
+        public String mName;
+        public int mErrCode;
+        public String mErrMsg;
+
+        public Erc731MetaData(String jsonString, int mErrCode, String mErrMsg) {
+            this.mErrCode = mErrCode;
+            this.mErrMsg = mErrMsg;
+            try {
+                JSONObject metaDataJObj = new JSONObject(jsonString);
```

Comment:
I would process the metadata only when error code is equal 0, otherwise the JSON string will be null.

Reply:
I think it's okay and concise as it's in `try` anyway to handle the mentioned use case. Although, `mErrCode` can be used to detect and display messages on NFTDetails screen.
Additionally, have tested the null use case, no issue.